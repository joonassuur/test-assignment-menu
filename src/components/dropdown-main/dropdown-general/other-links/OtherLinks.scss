.otherlinks-container {
  border-bottom: 1px solid #dcdce0;
  .item {
    display: flex;
    align-items: center;
    .material-icons-outlined {
      margin-right: 5px;
    }
  }
}
