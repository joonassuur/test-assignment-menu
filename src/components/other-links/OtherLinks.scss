.otherlinks-container {
  border-bottom: 1px solid #dcdce0;
}
