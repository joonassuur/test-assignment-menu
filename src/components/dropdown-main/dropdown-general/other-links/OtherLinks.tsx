import React from "react";
import { useDispatch } from "react-redux";
import { setDropdown } from "../../../../redux/Index";
import "./OtherLinks.scss";

function OtherLinks() {
  const dispatch = useDispatch();

  const closeDD = () => {
    dispatch(setDropdown(false));
  };

  return (
    <div className="otherlinks-container">
      <div className="item" onClick={closeDD}>
        <span className="material-icons-outlined">phone_iphone</span>
        <span>Get the mobile app</span>
      </div>
      <div className="item" onClick={closeDD}>
        <span className="material-icons-outlined">people</span>
        <span>Community</span>
      </div>
      <div className="item" onClick={closeDD}>
        <span className="material-icons-outlined">book</span>
        <span>Knowledge base</span>
      </div>
    </div>
  );
}

export default OtherLinks;
