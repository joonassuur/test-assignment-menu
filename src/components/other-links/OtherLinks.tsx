import React from "react";
import "./OtherLinks.scss";

function OtherLinks() {
  return (
    <div className="otherlinks-container">
      <div className="item">Get the mobile app</div>
      <div className="item">Community</div>
      <div className="item">Knowledge base</div>
    </div>
  );
}

export default OtherLinks;
