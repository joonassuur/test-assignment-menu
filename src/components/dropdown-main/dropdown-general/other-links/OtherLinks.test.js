import React from "react";
import TestRenderer from "react-test-renderer";

import OtherLinks from "./OtherLinks";

const mockDispatch = jest.fn();

jest.mock("react-redux", () => ({
  useSelector: jest.fn((fn) => fn()),
  useDispatch: () => mockDispatch,
}));

jest.mock("../../../../redux/Index");

describe("OtherLinks component", () => {
  test("Matches the snapshot", () => {
    const otherLinks = TestRenderer.create(<OtherLinks />);
    expect(otherLinks.toJSON()).toMatchSnapshot();
  });
});
