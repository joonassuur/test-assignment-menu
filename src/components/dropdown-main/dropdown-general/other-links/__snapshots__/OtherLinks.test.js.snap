// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OtherLinks component Matches the snapshot 1`] = `
<div
  className="otherlinks-container"
>
  <div
    className="item"
    onClick={[Function]}
  >
    <span
      className="material-icons-outlined"
    >
      phone_iphone
    </span>
    <span>
      Get the mobile app
    </span>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <span
      className="material-icons-outlined"
    >
      people
    </span>
    <span>
      Community
    </span>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <span
      className="material-icons-outlined"
    >
      book
    </span>
    <span>
      Knowledge base
    </span>
  </div>
</div>
`;
